apps/postgres-new/app/api/databases/[id]/upload/route.ts

```diff
@@ -37,15 +90,42 @@ export async function POST(req: NextRequest, { params }: { params: { id: string
 
   await upload.done()
 
-  return new Response(
-    JSON.stringify({
-      success: true,
-      data: {
-        serverName: `${databaseId}.${wildcardDomain}`,
-      },
-    }),
-    { headers: { 'content-type': 'application/json' } }
-  )
+  const { data: existingDeployedDatabase } = await supabase
+    .from('deployed_databases')
+    .select('id')
+    .eq('database_id', databaseId)
+    .maybeSingle()
+
+  let password: string | undefined
+
+  if (existingDeployedDatabase) {
+    await supabase
+      .from('deployed_databases')
+      .update({
+        deployed_at: 'now()',
+      })
+      .eq('database_id', databaseId)
+  } else {
+    password = generateDatabasePassword()
+    await supabase.from('deployed_databases').insert({
+      database_id: databaseId,
+      name,
+      created_at: createdAt,
+      auth_method: 'scram-sha-256',
+      auth_data: createScramSha256Data(password),
+    })
```

Comment:
Just a note for the future - if we're sticking with Supabase Storage, we could add referential integrity to `storage.objects` table which gets populated when objects are added to the bucket

Reply:
Do you mean like adding a fk to the storage table?

Reply:
Yes

Reply:
Issue here: https://github.com/supabase-community/postgres-new/issues/70